src/components/ViewCourse/sections/AddFastAddReserve.js

```diff
@@ -0,0 +1,72 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { Button, Icon } from '@folio/stripes/components';
+import { CalloutContext, Pluggable, stripesConnect } from '@folio/stripes/core';
+
+/*
+  This component relies entirely on the create-inventory-records plugin existing and nothing
+  will be rendered by `Pluggable` if it doesn't. The unit tests for this app are written such that
+  the thing being tested is a stripes-cli-served bundle of _just_ this app - there are no plugins!
+  As a result, any of the code inside of Pluggable (onClose, renderTrigger, etc) can't be executed
+  during tests so we ignore it from code-coverage calculations. This is suboptimal, but barring
+  significant changes to the testing framework (just go on ahead and tell Mike Taylor to consider
+  that) the best option we have for maintaining good enough coverage numbers is ignoring the block...
+  ...
+  ...this all smacks of Goodhart's Law don't it?
+*/
+/* istanbul ignore next */
+const AddFastReserve = ({
+  courseListingId,
+  mutator,
+}) => {
+  const callout = useContext(CalloutContext);
+
+  return (
+    <Pluggable
+      id="clickable-create-fast-add-inventory-records"
+      onClose={data => {
+        if (!data) return;
+
+        const { itemRecord } = data;
+        mutator.reserves.POST({ courseListingId, itemId: itemRecord.id })
+          .then(() => {
+            callout.sendCallout({ message: <FormattedMessage id="ui-courses.addItem.addedItem" values={{ title: itemRecord.title }} /> });
+          }).catch(exception => {
+            exception.text().then(text => {
+              callout.sendCallout('error', <FormattedMessage id="ui-courses.addItem.failure" values={{ message: text }} />);
+            });
+          });
+      }}
+      renderTrigger={({ onClick }) => (
```

Comment:
Not a review comment, just an idle question ... what does `renderTrigger` do? I've not seen it before, and it doesn't seem to be mentioned in `stripes-core/src/Pluggable.js`. I guess it's passed through to the plugin and used there?

Reply:
Exactly. `renderTrigger` is a pattern that we started using when in ERM plugins when increasing a11y and keyboard nav. We realised that we need to return focus to the triggering button after the plugin's modal is closed, but to do so we need a ref in the parent app. _That_ means the parent app either needs to control rendering or jump through hoops in the plugins and add more single-use props. So `renderTrigger` allows the parent app to be in full control. The Acquisitions teams picked up this approach as well in the meantime.

As a side benefit, it allows us to make our trigger be _exactly_ what we want it to be. In this case, the lightning icon hopefully helps to link "Fast Add" to the same concept in the Inventory app which also uses the same icon.

Reply:
Nice!

Reply:
Nice!